ci/scripts/min-deps.py

```diff
@@ -0,0 +1,102 @@
+#!python3
+from __future__ import annotations
+
+import argparse
+import sys
+from collections import deque
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+if sys.version_info >= (3, 11):
+    import tomllib
+else:
+    import tomli as tomllib
+
+from packaging.requirements import Requirement
+from packaging.version import Version
+
+if TYPE_CHECKING:
+    from collections.abc import Generator, Iterable
+
+
+def min_dep(req: Requirement) -> Requirement:
+    """
+    Given a requirement, return the minimum version specifier.
+
+    Example
+    -------
+
+    >>> min_dep(Requirement("numpy>=1.0"))
+    "numpy==1.0"
+    """
+    req_name = req.name
+    if req.extras:
+        req_name = f"{req_name}[{','.join(req.extras)}]"
+
+    # TODO: Should this be allowed?
```

Comment:
Hmm, for these, the semantics are “get the highest version compatible with all other restraints”.

And I think that there are some packages that we might want those semantics for.